zomes/hc_zome_trust_atom/src/trust_atom.rs

```diff
@@ -15,95 +15,147 @@ enum LinkDirection {
-      match value_decimal.round_sf_with_strategy(9, RoundingStrategy::MidpointAwayFromZero) {
-        Some(value_decimal) => {
-          if value_decimal == Decimal::ONE {
-            Ok(".999999999".to_string())
-          } else if value_decimal == Decimal::NEGATIVE_ONE {
-            Ok("-.999999999".to_string())
-          } else if value_decimal > Decimal::NEGATIVE_ONE && value_decimal < Decimal::ONE {
-            let value_zero_stripped = value_decimal.to_string().replace("0.", ".");
-            Ok(value_zero_stripped)
-          } else {
-            Err(WasmError::Guest(format!(
-              "Value must be in the range -1..1, but got: `{}`",
-              value_str
-            )))
+fn create_bucket_string(bucket_bytes: &[u8]) -> String {
+  let mut bucket = String::new();
+  for chunk in bucket_bytes {
+    let val = chunk;
+    bucket += (val % 10).to_string().as_str();
+  }
+  bucket
+}
+
+fn create_extra(input: BTreeMap<String, String>) -> ExternResult<String> {
+  let entry = Extra { fields: input };
+
+  create_entry(entry.clone())?;
+
+  let entry_hash_string = calc_extra_hash(entry)?.to_string();
+  Ok(entry_hash_string)
+}
+
+pub fn calc_extra_hash(input: Extra) -> ExternResult<EntryHash> {
+  let hash = hash_entry(input)?;
+  Ok(hash)
+}
+
+fn normalize_value(value_str: Option<String>) -> ExternResult<Option<String>> {
+  match value_str {
+    Some(value_str) => match Decimal::from_str(value_str.as_str()) {
+      Ok(value_decimal) => {
+        match value_decimal.round_sf_with_strategy(9, RoundingStrategy::MidpointAwayFromZero) {
+          Some(value_decimal) => {
+            if value_decimal == Decimal::ONE {
+              Ok(Some(".999999999".to_string()))
+            } else if value_decimal == Decimal::NEGATIVE_ONE {
+              Ok(Some("-.999999999".to_string()))
+            } else if value_decimal > Decimal::NEGATIVE_ONE && value_decimal < Decimal::ONE {
+              let value_zero_stripped = value_decimal.to_string().replace("0.", ".");
+              Ok(Some(value_zero_stripped))
+            } else {
+              Err(WasmError::Guest(format!(
+                "Value must be in the range -1..1, but got: `{}`",
+                value_str
+              )))
+            }
           }
+          None => Err(WasmError::Guest(format!(
+            "Value could not be processed: `{}`",
+            value_str
+          ))),
         }
-        None => Err(WasmError::Guest(format!(
-          "Value could not be processed: `{}`",
-          value_str
-        ))),
       }
+      Err(error) => Err(WasmError::Guest(format!(
+        "Value could not be processed: `{}`.  Error: `{}`",
+        value_str, error
+      ))),
+    },
+    None => Ok(None),
+  }
+}
+
+fn create_link_tag(link_direction: &LinkDirection, chunk_options: Vec<Option<String>>) -> LinkTag {
+  let mut chunks: Vec<String> = vec![];
+
+  for i in 0..chunk_options.len() {
+    if let Some(_) = chunk_options[i].clone() {
```

Comment:
```suggestion
    if let Some(chunk) = chunk_options[i].clone() {
```

Reply:
chunk would never get used in this block. no need to assign var